pkg/deployer/terraform/ensure.go

```diff
@@ -0,0 +1,215 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package terraform
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	apimacherrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	lsv1alpha1 "github.com/gardener/landscaper/pkg/apis/core/v1alpha1"
+	lsv1alpha1helper "github.com/gardener/landscaper/pkg/apis/core/v1alpha1/helper"
+	terraformv1alpha1 "github.com/gardener/landscaper/pkg/apis/deployer/terraform/v1alpha1"
+	"github.com/gardener/landscaper/pkg/deployer/terraform/terraformer"
+)
+
+const (
+	// RequeueAfter is the time after which to requeue the item.
+	ItemRequeue = 15 * time.Second
+
+	// DeadLinePodNotReady is the time to wait before deleting a not ready pod.
+	DeadLinePodNotReady = 10 * time.Minute
+)
+
+func (t *Terraform) Reconcile(ctx context.Context, op OperationType) (reconcile.Result, error) {
```

Comment:
this method looks a lot like the code in the container deploxer.
Maybe we can share some code for pod reconcilition etc.